airflow/www/static/js/dag/details/dagCode/index.tsx

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useRef } from "react";
+import { Box, Heading } from "@chakra-ui/react";
+import { useOffsetTop } from "src/utils";
+import Time from "src/components/Time";
+
+import { useDagCode } from "src/api";
+import CodeBlock from "./CodeBlock";
+
+const DagCode = () => {
+  const dagCodeRef = useRef<HTMLDivElement>(null);
+  const offsetTop = useOffsetTop(dagCodeRef);
+  const { lastParsedTime, codeSource = "" } = useDagCode();
```

Comment:
Let's also pull in `isLoading` and have a spinner.

Reply:
done